build_scripts/build_collect_diff

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+#
+# Script to build the kernel, collect compiled file lists using modified kernel scripts,
+# and generate a git diff report based on the collected lists.
+
+set -e
+
+# check and install gcc-11 if not already installed
+install_package_safe() {
+    if ! command -v libssl-dev &> /dev/null; then
+        sudo apt-get update
+        sudo apt-get install -y libssl-dev
+    else
+        echo "libssl-dev is already installed."
+    fi
+}
+
+# safely apply a patch to linux kernel
+apply_patch() {
+    # shellcheck disable=SC2154
+    local patch_path="$kernel_repo_path/scripts/change-impact-tools/fixdep-patch.file"
+
+    # Stash any changes if there is any
+    if ! git diff --quiet; then
+        echo "linux-clone has unstashed change. Please stash them and run the script again."
+        exit 1
+    fi
+
+    # Abort `git am` only if there is a patch being applied
+    if git am --show-current-patch &> /dev/null; then
+        echo "linux-clone has a patch being applied already. Should run git am --abort and try run the tool again."
+        exit 1
+    fi
+    echo "path check: $(pwd)"
+    git apply < "$patch_path"
+    echo "applied the git patch"
+    echo "path check: $(pwd)"
+}
+
+parse_source_json_file() {
+    local python_path="$kernel_repo_path/scripts/change-impact-tools/build_scripts/parse_json"
+    # shellcheck disable=SC2154
+    local cloned_repo_name="/$clone_dir/"
+    local input_path="$kernel_repo_path/scripts/change-impact-tools/build_data/compile_commands.json"
+    local output_path="$kernel_repo_path/scripts/change-impact-tools/build_data/sourcefile.txt"
+
+    ./"$python_path" "$cloned_repo_name" "$input_path" "$output_path"
+    display_file_head "$kernel_repo_path/scripts/change-impact-tools/build_data" "sourcefile.txt" 3
+}
+
+# generate the build file list after building the kernel
+generate_compiled_file_lists() {
+    # Generate compiled source files list
+    local json_output_path="$kernel_repo_path/scripts/change-impact-tools/build_data/compile_commands.json"
+    echo "path check: $(pwd)"
+    ./scripts/clang-tools/gen_compile_commands -o "$json_output_path"
```

Comment:
The script in the Linux kernel source tree still has the `.py` extension. I had to make this change to get your script to work properly.

```suggestion
    ./scripts/clang-tools/gen_compile_commands.py -o "$json_output_path"
```